packages/caravan-fees/src/cpfp.ts

```diff
@@ -0,0 +1,105 @@
+import { Transaction } from "bitcoinjs-lib-v5";
+import { unsignedMultisigTransaction } from "@caravan/bitcoin";
+import BigNumber from "bignumber.js";
+import { CPFPOptions, UTXO } from "./types";
+import { estimateVirtualSize } from "./utils";
+
+/**
+ * Prepare a CPFP (Child-Pays-for-Parent) transaction.
+ * This function doesn't broadcast the transaction, it just prepares it.
+ * @param options CPFP options including the parent transaction and new fee rate
+ * @returns A new transaction ready for signing and broadcasting
+ */
+export function prepareCPFPTransaction(options: CPFPOptions): Transaction {
+  const {
+    parentTransaction,
+    newFeeRate,
+    availableUTXOs,
+    destinationAddress,
+    network,
+    multisigDetails,
+  } = options;
+
+  // Find suitable output from parent transaction to spend
+  const parentOutput = findSuitableParentOutput(
+    parentTransaction,
+    availableUTXOs
+  );
+  if (!parentOutput) {
+    throw new Error("No suitable output found in parent transaction for CPFP");
+  }
+
+  const childSize = estimateChildTransactionSize({
+    inputCount: 1, // We're spending one output from the parent
+    outputCount: 2, // Destination output and potentially change
+    addressType: multisigDetails.addressType,
+    requiredSigners: multisigDetails.requiredSigners,
+    totalSigners: multisigDetails.totalSigners,
+  });
+  // Calculate required fee for both transactions
+  const combinedSize = parentTransaction.virtualSize() + childSize;
```

Comment:
This is a kind of nit, but writing it this way makes this package tightly coupled to a particular library (and version of the library)- bitcoinjs-lib. I.e. you can't use this function without also having a compatible version bitcoinjs-lib installed since you need to create this transaction object in order to pass it in to this function.

One way around this is to try and keep any functions or classes that are exported from the module to only take primitive value types as inputs. So for example, this function could take a `combinedSize` argument or a parentTx Size and childSize args as inputs. 

The API around this is admittedly tricky though. A class might be another way to handle it where it is initialized with primitive values (input and output counts/values, maybe a rawTx as hex as an input, etc.). Then the internals can convert those to bitcoinjs-lib Transaction value from which you can use the convenience methods. Alternatively, if you want to keep it functional (which matches the style of the code in caravan/bitcoin) you have entrypoint functions which take the raw values and then turn them in to the Transaction object and then continue on.

Reply:
resolved this with the new class approach and target on psbt instead of multi sig tx as we had earlier